tests/layer1/test_port_error.py

```diff
@@ -0,0 +1,91 @@
+import logging
+import pytest
+import random
+import time
+
+from tests.common.helpers.assertions import pytest_assert
+from tests.common.utilities import skip_release
+
+pytestmark = [
+    pytest.mark.disable_loganalyzer,  # disable automatic loganalyzer
+    pytest.mark.topology('any')
+]
+
+SUPPORTED_PLATFORMS = ["arista_7060x6"]
+
+
+class TestMACFault(object):
+    @pytest.fixture(autouse=True)
+    def is_supported_platform(self, duthost):
+        if any(platform in duthost.facts['platform'] for platform in SUPPORTED_PLATFORMS):
+            skip_release(duthost, ["201811", "201911", "202012", "202205", "202211", "202305", "202405"])
+        else:
+            pytest.skip("DUT has platform {}, test is not supported".format(duthost.facts['platform']))
+
+    @staticmethod
+    def get_mac_fault_count(dut, interface, fault_type):
+        output = dut.show_and_parse("show int errors {}".format(interface))
+        logging.info("Raw output for show int errors on {}: {}".format(interface, output))
+
+        fault_count = 0
+        for error_info in output:
+            if error_info['port errors'] == fault_type:
+                fault_count = int(error_info['count'])
+                break
+
+        logging.info("{} count on {}: {}".format(fault_type, interface, fault_count))
+        return fault_count
+
+    @staticmethod
+    def get_interface_status(dut, interface):
+        return dut.show_and_parse("show interfaces status {}".format(interface))[0].get("oper", "unknown")
+
+    @pytest.fixture
+    def select_random_interface(self, duthosts, enum_rand_one_per_hwsku_frontend_hostname):
+        dut = duthosts[enum_rand_one_per_hwsku_frontend_hostname]
+        interfaces = list(dut.show_and_parse("show interfaces status"))
```

Comment:
@vdahiya12 Can we check if the port has a transceiver connected to it?
You might also want to use conn_graph_facts to determine which ports are expected to have transceivers connected.

Reply:
maybe a better check would be to check if the link is already up on the testbed ?

Reply:
@vdahiya12 can we select atleast 5 random ports?